### PR TITLE
KAS-3584: Source files

### PR DESCRIPTION
**Pull request by sergiofenoll**
https://github.com/kanselarij-vlaanderen/app-kaleidos/pull/311
https://github.com/kanselarij-vlaanderen/yggdrasil/pull/43

Merges into #1464 because it's editing the same modal.

The edit modal is getting pretty large, if desired it can be put into a separate component to make the DocumentCard a little slimmer. If changes are okay, I can do the refactor before merging (didn't want to refactor without being sure it was okay so as to not refactor something that might need to change).

### CODE REVIEW

**Comment by erikap** [comment]
Hoort hier geen backend PR op app-kaleidos bij?

**Comment by sergiofenoll** [comment]
> Hoort hier geen backend PR op app-kaleidos bij?

zo'n kleine change dat ik het vergeten was te pushen :sweat_smile:

File: app/models/file.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ export default class File extends Model {
   @attr('string') extension;
   @attr('datetime') created;
 
-  @belongsTo('piece') piece;
+  @belongsTo('file', { inverse: null }) primarySource;
```
Comment (erikap) [inline app/models/file.js]: Is er een expliciete reden om inverse op `null` te zetten? Anders zou ik de inverse relatie wel specifiëren

File: app/components/documents/document-card.js [hunk 2]
```diff
@@ -213,31 +218,60 @@ export default class DocumentsDocumentCardComponent extends Component {
...
+    await this.uploadedSourceFile?.destroyRecord();
+    this.uploadedSourceFile = null;
+
+    this.isDeletingSourceFile = false;
   }
 
   @task
-  *savePieceName() {
+  *savePieceEdit() {
     const now = moment().toDate();
```
Comment (erikap) [inline app/components/documents/document-card.js]: Avoid moment if not necessary: `const now = new Date();`

File: app/components/documents/document-card.js [hunk 3]
```diff
@@ -213,31 +218,60 @@ export default class DocumentsDocumentCardComponent extends Component {
...
     this.piece.set('modified', now);
     this.piece.set('name', this.pieceNameBuffer);
```
Comment (erikap) [inline app/components/documents/document-card.js]: No need to use `set()`

**Review by erikap — COMMENTED** [review]
Some small comments. Ok for me to extract a separate component :ok_hand:

File: app/models/file.js (hunk 1 above)
Comment (sergiofenoll) [reply app/models/file.js]: Eigenlijk gewoon omdat de inverse relatie (nog) niet nodig is. Maar kan inderdaad gewoon toegevoegd worden.

**Comment by erikap** [comment]
Kan je hier een custom run voor opzetten @sergiofenoll ? Ik de backend-branch ook eerst development mergen.

**Comment by sergiofenoll** [comment]
http://kal-kastaar.s.redpencil.io:8080/job/kaleidos/job/custom_frontend_backend_combination/292/

**Comment by erikap** [comment]
http://kal-kastaar.s.redpencil.io:8080/job/kaleidos/job/custom_frontend_backend_combination_2/41/

**Comment by ValenberghsSven** [comment]
Previous jenkins run seems to have messed up (failing tests having nothing to do with this PR, but with a different PR)
It almost looks like we tried to check out our branch, but some other branch was used instead.
Seen it before, PR specific fails happening on other totally unrelated jenkins runs (no branching done).
Started a new run.
http://kal-kastaar.s.redpencil.io:8080/job/kaleidos/job/custom_frontend_backend_combination_2/42/
Edit: run green except for 1 flaky test

**Comment by ValenberghsSven** [comment]
http://kal-kastaar.s.redpencil.io:8080/job/kaleidos/job/custom_frontend_backend_combination_2/43/
1 failing test that got fixed in dev so technically green run.

**Pull request merged**